modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -166,6 +178,41 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def acceptFeeTxs(
+      balances: SortedMap[Address, Balance],
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): F[(SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]])] =
+      maybeTxs match {
+        case None => (balances, maybeTxs).pure[F]
+        case Some(txs) =>
+          val feeReferredAddresses = txs.flatMap(tx => Set(tx.value.source, tx.value.destination))
+          val feeReferredBalances = feeReferredAddresses.foldLeft(SortedMap.empty[Address, Long]) {
+            case (acc, address) =>
+              acc.updated(address, balances.getOrElse(address, Balance.empty).value.value)
+          }
+          val updatedFeeReferredBalances = txs
+            .foldLeft(feeReferredBalances) {
+              case (balances, tx) =>
+                balances
+                  .updatedWith(tx.source)(existing => (existing.getOrElse(Balance.empty.value.value) - tx.amount.value).some)
+                  .updatedWith(tx.destination)(existing => (existing.getOrElse(Balance.empty.value.value) + tx.amount.value).some)
+            }
+
+          updatedFeeReferredBalances.toList
+            .foldLeftM(SortedMap.empty[Address, Balance]) {
+              case (acc, (address, balance)) =>
+                NonNegLong
+                  .from(balance)
+                  .map(Balance(_))
+                  .map(acc.updated(address, _))
+                  .leftMap(e => new ArithmeticException(s"Unexpected state when applying fee transactions: $e"))
+                  .liftTo[F]
+            }
+            .map { updates =>
+              (balances ++ updates, txs.some)
```

Comment:
I think you are only accepting all or none, so possibly you don't need to pass fee txs to acceptance manager. You will either way throw an error if you can't accept a single fee tx.

Reply:
I left it just for consistency as all the `acceptXYZTransactions` return both balances and txs.